### PR TITLE
Error map for updateResults

### PR DESCRIPTION
**Pull request by msvinaykumar**
will get code with error message in updateResultsAPI

       "errors": {
                "409": "An entry for this record already exists!"
            },

### CODE REVIEW

File: src/main/java/com/autotune/database/service/ExperimentDBService.java [hunk 1]
```diff
@@ -231,6 +233,43 @@ public ValidationOutputData addRecommendationToDB(Map<String, KruizeObject> expe
...
+                            } else {
+                                LOGGER.debug("timestamp: {} notEnoughDataCount : {} recommendation size() : {} ", timestampKey, notEnoughDataCnt, recommendationHashMap.size());
+                            }
+                        }
+                    }
+                    emptyTimestamps.forEach((timestamp -> {
+                        cr.getData().remove(timestamp);
+                    }));  // remove timestamp with NOT_ENOUGH_DATA for all three category
+                }
+            }
```
Comment (dinogun) [inline src/main/java/com/autotune/database/service/ExperimentDBService.java:272]: This needs to be removed from here

Comment (msvinaykumar) [reply src/main/java/com/autotune/database/service/ExperimentDBService.java]: done

**Comment by dinogun** [comment]
The format that we have been using for all the APIs has been as below
```
{
  "message": "Mandatory parameters missing.",
  "httpcode": 400,
  "documentationLink": "",
  "status": "ERROR"
}
```
@msvinaykumar I think we need to follow the same format for each individual record inside the bulk JSON as well. This means that we don't need two different ways to handle error conditions, one type for the entire bulk record and one for individual. Instead we need to have the same format for both.

**Comment by msvinaykumar** [comment]
> The format that we have been using for all the APIs has been as below
> 
> ```
> {
>   "message": "Mandatory parameters missing.",
>   "httpcode": 400,
>   "documentationLink": "",
>   "status": "ERROR"
> }
> ```
> 
> @msvinaykumar I think we need to follow the same format for each individual record inside the bulk JSON as well. This means that we don't need two different ways to handle error conditions, one type for the entire bulk record and one for individual. Instead we need to have the same format for both.

ok let me check

**Comment by msvinaykumar** [comment]
> The format that we have been using for all the APIs has been as below
> 
> ```
> {
>   "message": "Mandatory parameters missing.",
>   "httpcode": 400,
>   "documentationLink": "",
>   "status": "ERROR"
> }
> ```
> 
> @msvinaykumar I think we need to follow the same format for each individual record inside the bulk JSON as well. This means that we don't need two different ways to handle error conditions, one type for the entire bulk record and one for individual. Instead we need to have the same format for both.

Done

File: src/main/java/com/autotune/analyzer/performanceProfiles/PerformanceProfileInterface/ResourceOptimizationOpenshiftImpl.java [hunk 2]
```diff
@@ -177,6 +174,44 @@ public void generateRecommendation(KruizeObject kruizeObject, List<ExperimentRes
...
+                                if (notEnoughDataCnt == recommendationHashMap.size()) {
+                                    emptyTimestamps.add(timestampKey); // found timestamp with NOT_ENOUGH_DATA for all three category
+                                } else {
+                                    LOGGER.debug("timestamp: {} notEnoughDataCount : {} recommendation size() : {} ", timestampKey, notEnoughDataCnt, recommendationHashMap.size());
+                                }
+                            }
+                        }
+                        emptyTimestamps.forEach((timestamp -> {
+                            cr.getData().remove(timestamp);
+                        }));  // remove timestamp with NOT_ENOUGH_DATA for all three category
```
[the remark this answers is not in the document]
Comment (msvinaykumar) [reply src/main/java/com/autotune/analyzer/performanceProfiles/PerformanceProfileInterface/ResourceOptimizationOpenshiftImpl.java:212]: sorry Dino I did not get . Do you want me to move this code somewhere else ?

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**